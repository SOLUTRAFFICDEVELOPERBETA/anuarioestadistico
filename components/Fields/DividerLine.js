import React from 'react';
import PropTypes from 'prop-types';
import { Divider, Tooltip } from '@material-ui/core';
import styled from '@emotion/styled';
import { DragPreviewImage, useDrag, } from 'react-dnd';
import { DragHandle } from '@material-ui/icons';
import { grey } from '@material-ui/core/colors';

// Componente con estilos para obtener un divider con color
const ColorDivider = styled.div`
    min-height: 8px;
    padding: 2px 0px;
    margin-bottom: 8px;
    transition: 0.5s ease;
    display: grid;
    grid-template-columns: 0px auto;
    align-items: center;

    .dragger {
        align-self: center;
        justify-self: center;
        width: 24px;
        height: 24px;
        visibility: hidden;
        opacity: 0;
        transition: visibility 0s, opacity 0.5s linear;
    }

    :hover .dragger {
        visibility: visible;
        opacity: 1;
    }

    .dragger:hover {
        cursor: move;
    }

    :hover {
        padding: 8px 0px;
        cursor: pointer;
        background-color: rgb(245, 245, 245);
        grid-template-columns: 30px auto;
    }

    :hover .MuiDivider-root {
        background-color: var(--primary);
    }
`;

/**
 * @description Componente del divisor para los documentos
 * @param {String} id
 * @param {Boolean} child
 * @param {func: () => void } onDelete
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 * consulte los elementos DragPreviewImage, useDrag en {@link https://react-dnd.github.io/react-dnd/docs/overview | React DnD Drag and Drop for React}
 */
const DividerLine = ({ id, child, onDelete }) => {
    const [{ isDragging }, drag, preview] = useDrag({
        item: { type: child ? 'CHILD' : 'ITEM', id },
        collect: (monitor) => ({
            isDragging: !!monitor.isDragging()
        })
    });

    if (isDragging) return null;

    return (
        <React.Fragment>
            <DragPreviewImage connect={preview} src="/static/img/DragPreview.png" />
            <Tooltip title="Click para Eliminar">
                <ColorDivider ref={drag} id={id} onClick={() => onDelete(id)}>
                    <span className="dragger">
                        <DragHandle fontSize="small" style={{ fill: grey[500] }} />
                    </span>
                    <Divider variant="middle" />
                </ColorDivider>
            </Tooltip>
        </React.Fragment>
    );
};

DividerLine.propTypes = {
    id: PropTypes.string.isRequired,
    child: PropTypes.bool,
    onDelete: PropTypes.func.isRequired
};

DividerLine.defaultProps = {
    child: false
};

export default DividerLine;
