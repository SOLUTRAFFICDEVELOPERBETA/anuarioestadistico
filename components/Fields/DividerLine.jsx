import React from 'react'
import PropTypes from 'prop-types'
import { Divider, Tooltip } from '@material-ui/core'
import styled from '@emotion/styled'
import { DragPreviewImage, useDrag } from 'react-dnd'
import { DragHandle } from '@material-ui/icons'
import { grey } from '@material-ui/core/colors'
// import clsx from 'clsx'

const ColorDivider = styled.div`
  min-height: 8px;
  padding: 2px 0px;
  margin-bottom: 8px;
  transition: 0.5s ease;
  display: grid;
  grid-template-columns: 0px auto;
  align-items: center;

    .dragger {
        align-self: center;
        justify-self: center;
        width: 24px;
        height: 24px;
        visibility: hidden;
        opacity: 0;
        transition: visibility 0s, opacity 0.5s linear;
    }

    :hover .dragger {
        visibility: visible;
        opacity: 1;
    }

    .dragger:hover {
        cursor: move;
    }

  :hover {
    padding: 8px 0px;
    cursor: pointer;
    background-color: rgb(245,245,245);
    grid-template-columns: 30px auto;
  }

  :hover .MuiDivider-root {
    background-color: var(--primary)
  }
`

const DividerLine = ({ id, child, onDelete }) => {
  const [{ isDragging }, drag, preview] = useDrag({
    item: { type: child ? 'CHILD' : 'ITEM', id },
    collect: monitor => ({
      isDragging: !!monitor.isDragging()
    })
  })

  if (isDragging) return null

  return (
    <React.Fragment>
      <DragPreviewImage connect={preview} src="/static/images/DragPreview.png" />
      <Tooltip title="Click para Eliminar">
        <ColorDivider ref={drag} id={id} onClick={() => onDelete(id)}>
          <span className="dragger">
            <DragHandle fontSize="small" style={{ fill: grey[500] }} />
          </span>
          <Divider variant="middle" />
        </ColorDivider>
      </Tooltip>
    </React.Fragment>
  )
}

DividerLine.propTypes = {
  id: PropTypes.string.isRequired,
  child: PropTypes.bool,
  onDelete: PropTypes.func.isRequired
}

DividerLine.defaultProps = {
  child: false
}

export default DividerLine
