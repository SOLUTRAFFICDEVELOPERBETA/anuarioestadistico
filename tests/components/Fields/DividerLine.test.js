import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import React from 'react';
import { DndProvider } from 'react-dnd'
import { HTML5Backend } from 'react-dnd-html5-backend'
import DividerLine from '../../../components/Fields/DividerLine';


describe('Pruebas en <DividerLine/>', () => {

    const onDelete = jest.fn();
    const wrapper = mount(
        <DndProvider backend={HTML5Backend}>
            <DividerLine
                id="divider-id"
                child={true}
                onDelete={onDelete}
            />
        </DndProvider>
    )
    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe llamar la funciÃ³n, onDelete()', () => {
        wrapper.find('div').prop('onClick')();

        expect(onDelete).toHaveBeenCalled();
        expect(onDelete).toHaveBeenCalledTimes(1)
    })

})