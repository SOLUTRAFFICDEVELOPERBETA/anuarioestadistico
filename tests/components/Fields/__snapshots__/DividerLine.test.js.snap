// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DividerLine/> debe mostrarse correctamente 1`] = `
Array [
  "",
  Array [
    <div
      aria-describedby={null}
      className=" css-10z2yyl-ColorDivider esb0tnc0"
      id="divider-id"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Click para Eliminar"
    >
      <span
        className="dragger"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          style={
            Object {
              "fill": "#9e9e9e",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
          />
        </svg>
      </span>
      <hr
        className="MuiDivider-root MuiDivider-middle"
      />
    </div>,
    "",
  ],
]
`;
